### PR TITLE
hypershift: propose pluggable konnectivity

### PR DESCRIPTION
**Pull request by 2uasimojo**
Add a proposal for a way to add any Service to the konnectivity-agent laid down by the hostedcontrolplane controller.

### CODE REVIEW

File: enhancements/hypershift/pluggable-konnectivity.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+At the time of this writing, the only (known) way to plug an admission webhook into this configuration includes standing up a separate konnectivity-agent with the ClusterIP of the Service pointing to the webhook server.
+
+It would be preferable to be able to plug such a Service into the existing konnectivity-agent already deployed by hypershift.
+- Save duplication: In the workaround, the konnectivity-agent deployment is identical (except for IPs) to hypershift's.
+- Save cluster resources: One konnectivity-agent workload instead of two.
+- Save etcd space: Particularly in use cases where etcd slots are at a premium, we can save three objects (Deployment, ReplicaSet, Pod) by consolidating.
+- Save devtest cycles and code: Developers don't need to figure out how to run a konnectivity-agent, or include one in their project.
+
+### User Stories
+
```
Comment (enxebre) [inline enhancements/hypershift/pluggable-konnectivity.md:51]: >I would like to be able to run a Service on behalf of a hosted control plane

could you be more specific about the user stories motivating this?

Comment (2uasimojo) [reply enhancements/hypershift/pluggable-konnectivity.md:51]: To be really specific, I'm coming from hive's Scale Mode feature ([card](https://issues.redhat.com/browse/HIVE-1862), [PR](https://github.com/openshift/hive/pull/1854) -- particularly [this commit](https://github.com/openshift/hive/pull/1854/commits/04bcf358ec1c6eae7b34edfb124e69c69669e3f9) where I would love to *not* need to package my own konnectivity-agent deployment).

Comment (slaviered) [reply enhancements/hypershift/pluggable-konnectivity.md:51]: The main known user story here is the case of zero worker hypershift mode with webhook support. In this case, we need to route the admission requests from the hosted control plane to the admission server pods on the hosting plane. 

The mechanism to achieve this includes adding a service and endpoint on the hosted plane that points to the admission service on the hosting plane ,and adding this clusterIP to the konnectivity agent.

The above mechanism is employed by many players such as hive, acm and assisted-installer - all of them are currently replicating the konnectivity agent.

Comment (zanetworker) [reply enhancements/hypershift/pluggable-konnectivity.md:51]: QQ: https://github.com/openshift/hypershift/pull/1865#issuecomment-1316800263

Comment (2uasimojo) [reply enhancements/hypershift/pluggable-konnectivity.md:51]: [Answered](https://github.com/openshift/hypershift/pull/1865#issuecomment-1317244899).

Comment (2uasimojo) [reply enhancements/hypershift/pluggable-konnectivity.md:51]: Adding that question & answer to the Risks and Mitigations section.

**Comment by 2uasimojo** [comment]
> @2uasimojo: The `lifecycle/frozen` label cannot be applied to Pull Requests.

Well then why did you tell me I could
> Exclude this proposal from closing by commenting /lifecycle frozen.

?

Silly bot.

**Pull request closed without merging**